### ISSUE TITLE
Bug: `useSprings` v9 fails to animate changing deps, report with a fix

### ISSUE THREAD

**Issue by bebbi** [react-spring/react-spring#714]
## 🐛 Bug Report

[This sandbox](https://codesandbox.io/s/naughty-hawking-xn7vq) should animate changed item heights every 2 seconds but it doesn't. A working fix is provided below.

## To Reproduce

Check out sandbox, see how items stay static

## Expected behavior

In the same sandbox, rename the `useSprings` hook call inside `DraggableList` to `useSpringsFixed` which is already imported.
--> Items now correctly change position every 2s.
The fix is based on a suggestion by @drcmda but my understanding is that `v9` should correctly animate this without further code.

## Link to repro (highly encouraged)

Sandbox above

## Environment

- `react-spring` v9.0.0-beta.23

**Comment by aleclarson** [comment]
Here's the problem line:
```js
const [springs, setSprings] = useSprings(items.length, fn(order, height), [items, height])
```
The `fn` function returns a function, which is only ever called once (on mount). We should probably change it to work with the `deps` argument.

I'll fix this when I get around to finishing and merging #670

**Comment by aleclarson** [comment]
This will be fixed by #808
